tests/test_flow/test_protocols.py

```diff
@@ -372,6 +372,26 @@ def np_array_high_d():
     assert(np.array_equal(arr, desired))
 
 
+def test_yaml_protocol_list(builder):
+    x = ['red', 'blue']
+    @builder
+    @bn.protocol.yaml(default_flow_style=False)
+    def alist():
+        return x
+
+    assert builder.build().get('alist') == x
+
+
+def test_yaml_protocol_dict(builder):
+    x = {'dumbo': {'token': 'payment_token', 'signals': ['abc', 'def']}}
+    @builder
+    @bn.protocol.yaml(encoding='utf-8')
```

Comment:
Similarly, if we want to test `encoding`, we probably want to either check that it actually took effect.